From 1073b690cb4effd2539b6a76df4c56b61ebd0856 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Mon, 11 Jun 2018 23:37:09 +0800
Subject: [PATCH] do not quote password/proxy_password

Upstream-Status: Inappropriate [pulsar specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 dnf/repo.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dnf/repo.py b/dnf/repo.py
index c897566..46d4ea7 100644
--- a/dnf/repo.py
+++ b/dnf/repo.py
@@ -81,7 +81,7 @@ def _user_pass_str(user, password):
     if user is None:
         return None
     user = dnf.pycomp.urllib_quote(user)
-    password = '' if password is None else dnf.pycomp.urllib_quote(password)
+    password = '' if password is None else password
     return '%s:%s' % (user, password)
 
 
-- 
2.7.4

